### PR TITLE
DashboardLinks: variables are resolved and limits to 100

### LINKED ISSUE #24994: Dashboard links not preserving variable values in 7.0.0
**What happened**:

On 6.7.3 we had dashboard links configured to preserve time and variable values when navigating between dashboards. Post upgrade to 7.0 these links no longer include any query parameters. 

I checked config and the options to preserve time range and variable values are still enabled, as described in https://grafana.com/docs/grafana/latest/linking/dashboard-links/

**What you expected to happen**:

Variable and time range selection to be preserved when clicking a dashbboard link.

**How to reproduce it (as minimally and precisely as possible)**:

- Add dashboard links to a dashboard, selecting the options to preserve time range and variables. 
- On the dashboard, select non-default variable values. 
- Click one of the dashboard links.
- Observe that the new dashboard uses its default variables, not those selected on the previous dashboard.

**Environment**:
- Grafana version: 7.0.0
- Data source type & version: Influx v1.4.3
- OS Grafana is installed on: Ubuntu 16.04
- User OS & Browser: Ubuntu 20.04, Firefox 76.0.1
- Grafana plugins: None
- Others:

### LINKED ISSUE #25019: Dashboard links only shows 7 dashboards in grafana 7.0.0
I have "many" sub dashboards that I link to from a main dashboard with the tag option of links.
Before upgrading to 7.0.0 I had all dashboard in the pulldown, now only 7.

**What you expected to happen**:
Show all dashboards with the named tag

**How to reproduce it (as minimally and precisely as possible)**:
Add multiple dashboards with the same tag and make a link that uses that tag to link to those dashboards.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.00
- Data source type & version: influx
- OS Grafana is installed on: Centos 7
- User OS & Browser: Chrome / firefox
- Grafana plugins:
- Others:
Screenshot_20200522_152632
Screenshot_20200522_152712

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This solves variables not being resolved when using dashboard links as a drop down or as a link. As this was the same area as #25019 I also changed the limit back to `100`.

**Which issue(s) this PR fixes**:
Fixes #24994
Fixes #25019

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 1]
```diff
@@ -42,8 +42,8 @@ export interface DashboardLink {
   type: DashboardLinkType;
   url: string;
   asDropdown: boolean;
-  tags: [];
-  searchHits?: [];
+  tags: any[];
+  searchHits?: any[];
```
Comment (hugohaggmark) [inline public/app/features/dashboard/state/DashboardModel.ts:46]: Weird typing before, don't know how that worked 🤔

File: public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx [hunk 2]
```diff
@@ -13,100 +13,80 @@ interface Props {
...
-              ...dash,
-              url: getLinkSrv().getLinkUrl(dash),
-            };
-          });
-
-        this.setState({
-          searchHits: processed,
-        });
-      });
+    const searchHits = await searchForTags(link);
```
Comment (dprokop) [inline public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx:31]: The limit = 7 was suuuuper weird. Is this a mistake or was it like this before migration? :D

Comment (hugohaggmark) [reply public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx:31]: A mistake while migrating to React it seems.

**Review by dprokop — APPROVED** [review]
This is looking great, with the tests and all the glitter 👍

**Comment by hugohaggmark** [comment]
> This is looking great, with the tests and all the glitter 👍

Thanks @dprokop

**Pull request merged**